go/client/cmd_version.go

```diff
@@ -63,18 +71,58 @@ func (v *CmdVersion) ParseArgv(c *cli.Context) error {
 		v.mode = modeVerbose
 	}
 	v.svc = !c.Bool("S")
+	v.assertMatching = c.Bool("assert-matching")
+	if v.assertMatching && (v.mode != modeNormal || !v.svc) {
+		return errors.New("cannot use --assert-matching with other options")
+	}
 	return nil
 }
 
 func (v *CmdVersion) Run() error {
 	var err error
+	if v.assertMatching {
+		return v.runAssertMatching()
+	}
 	v.runLocal()
 	if v.svc {
 		err = v.runService()
 	}
 	return err
 }
 
+func (v *CmdVersion) runAssertMatching() error {
+	cli, err := GetConfigClient(v.G())
+	if err != nil {
+		v.G().Log.Debug("no service running: %v", err)
+		return nil
```

Comment:
are you sure this is what you want?

Reply:
What happens if the service is not running, but KBFS or the gui are?

Reply:
If that's happening, which sucks, there's no way to get the kbfs/gui version without directly connecting to them, which I'm not sure how to do (and may not be possible currently, but I asked patrick).

Reply:
You could have this CLI process directly connect to them if the service is not found/not running. Or spawn a temporary service for that reason?